admin/services/pin-generation.service.js

```diff
@@ -63,12 +63,10 @@ pinGenerationService.getPupils = async (dfeNumber, sortField, sortDirection) =>
  * @returns {Promise<*>}
  */
 pinGenerationService.filterGroups = async (schoolId, pupils) => {
-  let pupilIds = ''
-  if (pupils.length > 0) {
-    pupils.map(p => { if (p.group_id) pupilIds += "'" + p.group_id + "', " })
-    pupilIds = pupilIds.length > 0 ? pupilIds.substr(0, pupilIds.length - 2) : ''
+  if (pupils.length < 1) {
+    return false
   }
-  return groupDataService.sqlFindGroupsByIds(schoolId, pupilIds)
+  return groupDataService.sqlFindGroupsByIds(schoolId, pupils)
```

Comment:
It looked better to me as pupilId.   Is the whole pupil object required?  Why?

Reply:
Renamed. 
Please see previous explanation regarding the argument passed.